diff --git a/src/httpclient.coffee b/src/httpclient.coffee
index bd433c1..6956403 100644
--- a/src/httpclient.coffee
+++ b/src/httpclient.coffee
@@ -24,7 +24,7 @@ http_request = (options , params , callback) ->
   if params and options.method == "POST"
     data = new Buffer jsons params, 'utf8'
     options.headers["Content-Type"] = config.contentType
-    options.headers["Content-Length"] = data.byteLength
+    options.headers["Content-Length"] = Buffer.byteLength data.toString(), 'utf8'
     options.headers["Connection"] = config.connection
   if params and options.method == "GET"
     query = querystring.stringify params
